addons/survey/views/survey_templates.xml

```diff
@@ -185,10 +185,21 @@
             t-att-data-timer="timer_start"
             t-att-data-time-limit-minutes="time_limit_minutes"/>
         <t t-if="survey.questions_layout == 'one_page'">
-            <t t-foreach='survey.question_and_page_ids' t-as='question'>
-                <h2 t-if="question.is_page" t-field='question.title' class="o_survey_title pb16 text-break w-lg-50 mx-lg-auto" />
-                <div t-if="question.is_page" t-field='question.description' class="text-break w-lg-50 mx-lg-auto"/>
-                <t t-if="not question.is_page and question in answer.predefined_question_ids" t-call="survey.question_container"/>
+            <!-- Questions without section -->
+            <t t-foreach="survey.question_ids.filtered(lambda q: not q.page_id)" t-as="question">
+                <t t-if="question in answer.predefined_question_ids" t-call="survey.question_container"/>
+            </t>
+            <!-- Questions with section -->
+            <t t-foreach="survey.page_ids" t-as="page">
+                <t t-set="display_section" t-value="any(not q.triggering_answer_ids for q in page.question_ids)
+                    or (page.question_ids and survey.questions_selection == 'random')"/>
+                <div t-attf-class="js_section_wrapper #{'d-none' if not display_section else ''}">
+                    <h2 t-field='page.title' class="o_survey_title pb16 text-break w-lg-50 mx-lg-auto" />
+                    <div t-field='page.description' class="text-break w-lg-50 mx-lg-auto"/>
+                    <t t-foreach="page.question_ids" t-as="question">
+                        <t t-if="question in answer.predefined_question_ids" t-call="survey.question_container"/>
+                    </t>
+                </div>
```

Comment:
Hi @flch-odoo :slightly_smiling_face: 
I'm doing a task where I need to hide the empty sections (the ones without any question to display) in the one_page layout as it is resulting in strange display like so:
(sections title and description displayed even though the sections are empty because of conditional questions)
![image](https://github.com/odoo/odoo/assets/107679666/713c8f83-819b-499c-b03e-fee9792d1bbb)
In that matter, @julien-banken made a suggestion here https://github.com/odoo/odoo/pull/149060#discussion_r1469690930 which I implemented and it seems to be working fine :slightly_smiling_face: 

But I'm not entirely sure that this won't break anything in certain specific cases :thinking:
Can I have your opinion on this? :pray: 
Thanks!